components/server/src/prebuilds/incremental-workspace-service.ts

```diff
@@ -78,79 +82,124 @@ export class IncrementalWorkspaceService {
             return;
         }
 
-        const imageSourcePromise = this.imageSourceProvider.getImageSource({}, user, context, config);
-
-        // Note: This query returns only not-garbage-collected prebuilds in order to reduce cardinality
-        // (e.g., at the time of writing, the Gitpod repository has 16K+ prebuilds, but only ~300 not-garbage-collected)
-        const recentPrebuilds = await this.workspaceDB.findPrebuildsWithWorkspace(projectId);
-        const imageSource = await imageSourcePromise;
-        for (const recentPrebuild of recentPrebuilds) {
-            if (
-                this.isGoodBaseforIncrementalBuild(
-                    history,
-                    config,
-                    imageSource,
-                    recentPrebuild.prebuild,
-                    recentPrebuild.workspace,
-                    includeUnfinishedPrebuilds,
-                )
-            ) {
-                return recentPrebuild.prebuild;
+        const [recentPrebuilds, imageSource] = await Promise.allSettled([
```

Comment:
I did this because both queries (the image one which reaches out to the external SCM API and the prebuild one doing potentially big DB lookups) can take some time. There doesn't seem to be good reason to not do both at once.